packages/field/src/useField.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { generateId } from './utils/IdManager';
+
+export function useField(idPrefix) {
+  const id = idPrefix || generateId('garden-field-container');
```

Comment:
I think there might be a problem here. Whenever `useField` is re-rendered this will generate a new ID each time.

A storybook example that raises state out of a component that's using `useField` (like a `TextInput` component) will increment the ID every value change.

```jsx
const Field = ({ id }) => {
      const { getLabelProps, getInputProps, getHintProps } = useField(id);
      const [value, setValue] = React.useState('');

      return (
        <>
          <label {...getLabelProps()}>Accessible Native Input</label>
          <div {...getHintProps()}>Optional Hint</div>
          <input {...getInputProps({ value, onChange: e => setValue(e.target.value) })} />
        </>
      );
    };
```

Either of the following should store the value between renders:

```jsx
const [id] = useState(idPrefix || generateId());

const idRef = useRef(idPrefix || generateId());
const id = idRef.current;
```

Reply:
yeah good catch.